Wallix

yoennb opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0


## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=25840) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=25840) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=25840) it.</sub>

edik24 commented:
Hi @yoennb,
can you please [sign CLA](https://github.com/demisto/content/pull/25840#issuecomment-1508222422),
and fill the [contribution form](https://github.com/demisto/content/pull/25840#issuecomment-1508224125)?

yoennb commented:
We are working on it. We first need to finalize the video required to sign the contribution form.

samuelFain commented:
Hi @yoennb, thank you for contributing!
I'm currently reviewing your PR and should have a review ready shortly.
@ShirleyDenkberg, can you please review?

samuelFain commented on Packs/WALLIXBastion/Integrations/WAB/WAB.py:
```diff
@@ -0,0 +1,2724 @@
+
+
+def test_module(client: Client) -> None:
+    # Test functions here
```
Dont forget to complete the test_module function.

samuelFain commented on Packs/WALLIXBastion/Integrations/WAB/WAB.yml:
```diff
@@ -0,0 +1,4725 @@
+description: Centralized Control and Monitoring of Privileged Access to Sensitive Assets
+fromversion: 6.10.0
+tests:
+  - No test - todo
```
Make sure do update the `tests` key if needed.

samuelFain reviewed (commented):
@yoennb, Great Job!
Please see my comments, and make sure to merge from master and handle all the linter errors (flake8 etc.).
Also, don't forget to share the demo video with me when its available.

yoennb commented:
Hello @samuelFain I have updated the PR according to your suggestions and added a few authentication handling code needed on our side. We are working on the video.

Also, by "merge from master" you mean rebase our branch on your master branch ?

samuelFain reviewed (commented):
Hi @yoennb, please see these comments so that we may pass the build checks.
Feel free to share your video with me when its available 👍

samuelFain commented on Packs/WALLIXBastion/TestPlaybooks/WAB_test.yml:
Please add the missing `fromversion` field.

samuelFain reviewed (commented):
@yoennb, great work!
Added a few minor comments to fix validation and linter errors.
Let me know when the demo video is ready, you can share it with me via my email sfainberg@paloaltonetworks.com.

samuelFain commented on Packs/WALLIXBastion/Integrations/WAB/WAB.py:
```diff
@@ -0,0 +1,2810 @@
+        )  # no AuthError if first_auth is True
+
+        try:
+            resp: Response = super()._http_request(
+                *args, **kwargs, resp_type="response", error_handler=client_err_handler
+            )
```
I don't mind if you believe it's not an issue with the code itself that can cause further problems.

samuelFain commented:
Hi @yoennb,
This is a kindly reminder regarding the demo video.
If you prefer, we can schedule a meeting instead and do a live demo, whatever works best for you.

samuelFain reviewed (commented):
@yoennb Great work, we are almost done here...
There are some last issues to address before we merge this PR:
- Please see my comment on updating the integration category in the yml file.
- There are some Flake8 and Mypy errors thats failing the linting checks, see them [here](https://app.circleci.com/pipelines/github/demisto/content/257774/workflows/502582ec-36c7-46b9-b253-b9f882c8938c/jobs/699568).

Let me know if something is unclear. 
BTY, the demo video was excellent!

yoennb commented:
Thanks! I'm on it

yoennb commented:
@samuelFain "contribution form filled label verification" --> should I fill the form again?

samuelFain commented:
@yoennb, is it possible that the CLA is not filled yet? if so, please fill it as this PR can't be merged without it.
> Hi @yoennb, can you please [sign CLA](https://github.com/demisto/content/pull/25840#issuecomment-1508222422), and fill the [contribution form](https://github.com/demisto/content/pull/25840#issuecomment-1508224125)?

Also, we noticed the forked repo is named with `demisto` instead of the suggested `content` format.
This current situation is preventing us to finish the merging process.
Could you please manually rename the repo using this manual?
https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository

Please notify me if there are any questions.
Thanks!

yoennb commented:
@samuelFain the CLA is already signed, I cannot sign it again. I tried filling the contribution form again just in case.

samuelFain commented:
Merged 👍

yoennb commented:
Excellent! Glad we managed to work this out. Should I request the renaming of our forked repo anyway?
